### PR TITLE
[ENG-11658][eas-cli] Add --no-channel flag for update and rollback to embedded

### PR DESCRIPTION
**Pull request by quinlanj**
# Why

When users create an update, we make sure that the branch they publish to is linked to a channel of the same name. However, this behaviour is not always desirable. For example, in the development usecase where someone is trying to load their update in the dev-client, a channel is not necesary. Users will either navigate to the update they want in the dev client menu or via a preview QR code. 

When we've interviewed users who use the dev client/qr code flow, they are always running `eas delete:channel` after the `eas update` command. 

# How

This PR adds a flag, `--no-channel` for the development usecase. This prevents a channel from being automatically created. 

# Test Plan

- [ ] new tests pass

### CODE REVIEW

**Comment by quinlanj** [comment]
/changelog-entry new-feature Add --no-channel flag for update and rollback to embedded

File: packages/eas-cli/src/commands/update/index.ts [hunk 1]
```diff
@@ -143,6 +145,10 @@ export default class UpdatePublish extends EasCommand {
       description: `File containing the PEM-encoded private key corresponding to the certificate in expo-updates' configuration. Defaults to a file named "private-key.pem" in the certificate's directory.`,
       required: false,
     }),
+    'no-channel': Flags.boolean({
+      description: `Skip EAS channel creation.`,
+      default: false,
+    }),
```
Comment (szdziedzic) [inline packages/eas-cli/src/commands/update/index.ts:151]: maybe we can do
```suggestion
    'no-channel': Flags.boolean({
      description: `Skip EAS channel creation.`,
      default: false,
      exclusive: ['channel'],
    }),
```
?

File: packages/eas-cli/src/commands/update/index.ts [hunk 2]
```diff
@@ -548,6 +555,10 @@ export default class UpdatePublish extends EasCommand {
       );
     }
 
+    if (channelName && flags['no-channel']) {
+      Errors.error('Cannot use --channel with --no-channel', { exit: 1 });
+    }
```
Comment (szdziedzic) [inline packages/eas-cli/src/commands/update/index.ts:560]: ditto

File: packages/eas-cli/src/commands/update/roll-back-to-embedded.ts [hunk 3]
```diff
@@ -104,6 +106,10 @@ export default class UpdateRollBackToEmbedded extends EasCommand {
       description: `File containing the PEM-encoded private key corresponding to the certificate in expo-updates' configuration. Defaults to a file named "private-key.pem" in the certificate's directory.`,
       required: false,
     }),
+    'no-channel': Flags.boolean({
+      description: `Skip EAS channel creation.`,
+      default: false,
+    }),
```
Comment (szdziedzic) [inline packages/eas-cli/src/commands/update/roll-back-to-embedded.ts:112]:
```suggestion
    'no-channel': Flags.boolean({
      description: `Skip EAS channel creation.`,
      default: false,
      exclusive: ['channel'],
    }),
```

File: packages/eas-cli/src/commands/update/roll-back-to-embedded.ts [hunk 4]
```diff
@@ -387,13 +395,18 @@ export default class UpdateRollBackToEmbedded extends EasCommand {
       );
     }
 
+    if (channelName && flags['no-channel']) {
+      Errors.error('Cannot use --channel with --no-channel', { exit: 1 });
+    }
```
Comment (szdziedzic) [inline packages/eas-cli/src/commands/update/roll-back-to-embedded.ts:400]: ditto

**Review by szdziedzic — APPROVED** [review]

**Review by wschurman — APPROVED** [review]
Agree with @szdziedzic's suggestions.

**Comment by quinlanj** [comment]
closing in favour of https://github.com/expo/eas-cli/pull/2346 , appreciate the suggestions tho, @szdziedzic !

**Pull request closed without merging**